libstuff/STCPManager.cpp

```diff
@@ -8,218 +8,196 @@
 
 atomic<uint64_t> STCPManager::Socket::socketCount(1);
 
-STCPManager::~STCPManager() {
-    SASSERTWARN(socketList.empty());
-}
-
-void STCPManager::prePoll(fd_map& fdm) {
-    // Add all the sockets
-    for (Socket* socket : socketList) {
-        // Make sure it's not closed
-        if (socket->state.load() != Socket::CLOSED) {
-            // Check and see if it looks like we're still valid.
-            if (socket->s < 0) {
-                SWARN("Invalid FD number("
-                      << socket->s << "), we're probably about to corrupt stack memory. FD_SETSIZE=" << FD_SETSIZE);
+void STCPManager::prePoll(fd_map& fdm, Socket& socket) {
```

Comment:
Instead of running through the whole list of everything in the manager, we only want to poll specific commands, which manage their own lists of transactions, which are not all the transactions for a specific manager, and may be in several managers. So these just get called for each transaction one at a time, and then the single socket per transaction.

Otherwise, this isn't too different, it takes a single socket as an argument instead of looping through socketList.